server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -76,64 +79,78 @@ void setMessageListener(TransportMessageListener listener) {
         }
     }
 
+    void setSlowLogThreshold(TimeValue slowLogThreshold) {
+        this.slowLogThresholdMs = slowLogThreshold.getMillis();
+    }
+
     void inboundMessage(TcpChannel channel, InboundMessage message) throws Exception {
-        channel.getChannelStats().markAccessed(threadPool.relativeTimeInMillis());
+        final long startTime = threadPool.relativeTimeInMillis();
+        channel.getChannelStats().markAccessed(startTime);
         TransportLogger.logInboundMessage(channel, message);
 
         if (message.isPing()) {
             keepAlive.receiveKeepAlive(channel);
         } else {
-            messageReceived(channel, message);
+            messageReceived(channel, message, startTime);
         }
     }
 
-    private void messageReceived(TcpChannel channel, InboundMessage message) throws IOException {
+    private void messageReceived(TcpChannel channel, InboundMessage message, long startTime) throws IOException {
         final InetSocketAddress remoteAddress = channel.getRemoteAddress();
         final Header header = message.getHeader();
         assert header.needsToReadVariableHeader() == false;
 
-        if (header.isRequest()) {
-            handleRequest(channel, header, message);
-        } else {
-            // Responses do not support short circuiting currently
-            assert message.isShortCircuit() == false;
-            final TransportResponseHandler<?> handler;
-            long requestId = header.getRequestId();
-            if (header.isHandshake()) {
-                handler = handshaker.removeHandlerForHandshake(requestId);
+        try {
+            if (header.isRequest()) {
+                handleRequest(channel, header, message);
             } else {
-                TransportResponseHandler<? extends TransportResponse> theHandler =
-                    responseHandlers.onResponseReceived(requestId, messageListener);
-                if (theHandler == null && header.isError()) {
+                // Responses do not support short circuiting currently
+                assert message.isShortCircuit() == false;
+                final TransportResponseHandler<?> handler;
+                long requestId = header.getRequestId();
+                if (header.isHandshake()) {
                     handler = handshaker.removeHandlerForHandshake(requestId);
                 } else {
-                    handler = theHandler;
-                }
-            }
-            // ignore if its null, the service logs it
```

Comment:
git diff is not really clear, what I actually did is wrapping the whole method in a try block and added finally to align with linked patch https://github.com/elastic/elasticsearch/commit/5112c173194b70d2ed17f2ed3071dc0c9a44b76e#diff-e7af60ad8aea4d9c61ee4ae02164f46abee2943b8bc93c88161a9f03276c2cbfR149-R155